File: services/core-go/internal/core/domain/notification.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

type NotificationSeverity string

const (
	NotificationSeverityCritical NotificationSeverity = "CRITICAL"
	NotificationSeverityWarning  NotificationSeverity = "WARNING"
	NotificationSeverityInfo     NotificationSeverity = "INFO"
)

type NotificationCategory string

const (
	NotificationCategoryInvoice  NotificationCategory = "INVOICE"
	NotificationCategoryCashflow NotificationCategory = "CASHFLOW"
	NotificationCategoryVendor   NotificationCategory = "VENDOR"
	NotificationCategoryShipment NotificationCategory = "SHIPMENT"
	NotificationCategorySystem   NotificationCategory = "SYSTEM"
)

type Notification struct {
	ID         uuid.UUID            `json:"id" db:"id"`
	UserID     *uuid.UUID           `json:"user_id,omitempty" db:"user_id"`
	TargetRole *string              `json:"target_role,omitempty" db:"target_role"`
	Title      string               `json:"title" db:"title"`
	Message    string               `json:"message" db:"message"`
	Category   NotificationCategory `json:"category" db:"category"`
	Severity   NotificationSeverity `json:"severity" db:"severity"`
	ActionURL  *string              `json:"action_url,omitempty" db:"action_url"`
	Metadata   *string              `json:"metadata,omitempty" db:"metadata"`
	IsRead     bool                 `json:"is_read" db:"is_read"`
	ReadAt     *time.Time           `json:"read_at,omitempty" db:"read_at"`
	CreatedAt  time.Time            `json:"created_at" db:"created_at"`
}

File: apps/web-next/src/types/notification.ts
export type NotificationSeverity = "CRITICAL" | "WARNING" | "INFO";
export type NotificationCategory = "INVOICE" | "CASHFLOW" | "VENDOR" | "SHIPMENT" | "SYSTEM";

export interface NotificationItem {
  id: string;
  user_id: string;
  title: string;
  message: string;
  severity: NotificationSeverity;
  category: NotificationCategory;
  action_url?: string;
  is_read: boolean;
  read_at?: string | null;
  created_at: string;
}

export interface UnreadCountResponse {
  unread_count: number;
}
